price-estimator/src/orderbook.rs

```diff
@@ -40,8 +43,12 @@ impl Orderbook {
             }),
             // TODO: pass this from command line argument
             extra_rounding_buffer_factor: 2.0,
-            // TODO: pass real token infos, and update it with a real price source in a future PR
-            infallible_price_source: InfalliblePriceSource::new(HashMap::new()),
+            // TODO: pass real token infos
+            infallible_price_source: RwLock::new(InfalliblePriceSource::new(HashMap::new())),
```

Comment:
Looking at all the dependencies needed for the the InfalliblePriceSource, I'm wondering if the Infallible price source should be in charge of updating itself given a pricegraph (and own the external prices sources and all_tokens itself). 
Looking at the TODOs here it might be worthwhile to extract the "creation" logic somewhere outside the orderbook and make the pricesource a parameter for new.

Reply:
Implemented this suggestion.